app/src/main/java/org/greenstand/android/TreeTracker/viewmodels/NewTreeViewModel.kt

```diff
@@ -68,7 +70,10 @@ class NewTreeViewModel(
         val deltaSteps = absoluteStepCount - lastStepCountWhenCreatingTree
         newTree.addTreeAttribute(Tree.ABS_STEP_COUNT_KEY, absoluteStepCount.toString())
         newTree.addTreeAttribute(Tree.DELTA_STEP_COUNT_KEY, deltaSteps.toString())
-
+        deviceOrientation.rotationMatrixSnapshot?.let {
```

Comment:
I modeled the rotationMatrixSnapshot as a nullable value because the called here has no knowledge whether a invocation of takeSnapshotAndDisable method is called or not in DeviceOrientation. The nullability takes the guessing out.